libxrdp/xrdp_rdp.c

```diff
@@ -160,9 +163,45 @@ xrdp_rdp_read_config(struct xrdp_client_info *client_info)
                 client_info->use_fast_path = 0;
             }
         }
-        else if (g_strcasecmp(item, "disableSSLv3") == 0)
+        else if (g_strcasecmp(item, "ssl_protocols") == 0)
         {
-            client_info->disableSSLv3 = g_text2bool(value);
+            /* put leading/trailing space to properly detect "TLSv1" without regex */
+            tmp_length = g_strlen(value) + 3;
+            tmp = g_new(char, tmp_length);
+            g_snprintf(tmp, tmp_length, "%s%s%s", " ", value, " ");
+
+            /* disable all protocols first, enable later */
+            client_info->ssl_protocols =
+                SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1 | SSL_OP_NO_TLSv1_2;
+
+            if (g_pos(tmp, " TLSv1.2 ") >= 0)
```

Comment:
Yes probably we can but I want to use regex to parse comma separated value  as "TLSv1" is substring of "TLSv1.1". The current implementation correctly detects "TLSv1" without regex.

See also discussions in this tree https://github.com/neutrinolabs/xrdp/pull/542#discussion_r92333250 .